SPARQLWrapper/main.py

```diff
@@ -136,18 +135,27 @@ def main(test=None):
     sparql.setReturnFormat(args.format)
     results = sparql.query().convert()
 
-    if args.format == "json":
+    ct = type(results)
+    if ct is dict:
+        # "json"
         print(json.dumps(results, indent=4))
-    elif args.format in ("xml", "rdf+xml", "json-ld"):
-        print(results.toxml())  # type: ignore
-    elif args.format == "n3":
-        g = rdflib.Graph()
-        g.parse(data=results, format="n3")  # type: ignore
-        print(g.serialize(format="n3"))
-    elif args.format in ("csv", "tsv", "turtle"):
-        print(results.decode("utf-8"))  # type: ignore
+    elif ct is xml.dom.minidom.Document:
+        # "xml", "rdf+xml", "json-ld"
+        print(results.toxml())
+    elif ct is bytes:
+        if args.format == "n3":
+            g = rdflib.Graph()
+            g.parse(data=results, format="n3")
+            print(g.serialize(format="n3"))
+        else:
+            # "csv", "tsv", "turtle"
+            print(results.decode("utf-8"))
+    elif ct is rdflib.graph.ConjunctiveGraph:
```

Comment:
```suggestion
    elif isinstance(result, rdflib.graph.ConjunctiveGraph):
```

Reply:
Also, it may be best to just make it `rdflib.graph.Graph` if you are not using any ConjunctiveGraph specific methods, but as far as I can tell from the code it should always be ConjunctiveGraph so I'm fine if you leave it like this.

```suggestion
    elif isinstance(result, rdflib.graph.Graph):
```

Reply:
There is one caveat here also, if the result contains quads, some information will likely get lost as the default format is turtle. See https://github.com/RDFLib/rdflib/issues/1674